### ISSUE TITLE
Typescript compilation error

### ISSUE THREAD

**Issue by doomsower** [ZacharyRSmith/react-router-navigation-prompt#40]
**Do you want to request a *feature* or report a *bug*?**
Bug
**What is the current behavior?**
Typescript compilation fails with:
```bash
$ tsc
node_modules/react-router-navigation-prompt/types/index.d.ts:44:66 - error TS2304: Cannot find name 'NavigationPromptProps'.

44 export default function NavigationPrompt(): React.Component<Omit<NavigationPromptProps, keyof RouteComponentProps<any>>>;
                                                                    ~~~~~~~~~~~~~~~~~~~~~
```
**If the current behavior is a bug, please provide the steps to reproduce.**

Try to compile project that uses `react-router-navigation-prompt` with typescript 3.1.1 with this tsconfig:
```json
{
  "compilerOptions": {
    "noEmit": true,
    "noEmitOnError": true,
    "allowSyntheticDefaultImports": true,
    "esModuleInterop": true,
    "declaration": false,
    "emitDecoratorMetadata": true,
    "experimentalDecorators": true,
    "forceConsistentCasingInFileNames": true,
    "importHelpers": true,
    "noEmitHelpers": true,
    "jsx": "react",
    "lib": ["esnext", "dom"],
    "module": "esnext",
    "target": "es5",
    "moduleResolution": "node",

    "skipLibCheck": false,

    "strict": true,
    "strictFunctionTypes": false,

    "pretty": true
  }
}

```

**What is the expected behavior?**

Should compile without errors.

**Please post the output of `npm list react react-router react-router-dom react-router-navigation-prompt` (with redactions if needed).**
```bash
├─┬ react-images@1.0.1-alpha.1
│ └─┬ react-full-screen@0.2.3
│   └── react@15.6.2 
├── react-router@4.3.1 
├─┬ react-router-dom@4.3.1 
│ └── react-router@4.3.1  deduped
└── react-router-navigation-prompt@1.7.0 
```

**Comment by doomsower** [comment]
Probably `export default function NavigationPrompt(): React.Component<Omit<NavigationPromptProps, keyof RouteComponentProps<any>>>;` should be inside `declare module`, but there is a name conflict. Someone with knowledge of export intentions (what should be exported and under which name) should solve it

**Comment by ZacharyRSmith** [comment]
Thanks for the issue, @doomsower !  Unfortunately, I have little experience with Typescript and I do not have the time to look into this right now.

@bparish628, I'd appreciate it if you could look into this!

**Comment by ZacharyRSmith** [comment]
@doomsower , @bparish628 fixed, and it is published as v1.7.1
